app/models/host/managed.rb

```diff
@@ -944,4 +946,19 @@ def send_built_notification
   rescue SocketError, Net::SMTPError => e
     Foreman::Logging.exception("Host has been created. Failed to send email", e)
   end
+
+  # Ensures that object assigned in the association belongs to the taxonomies of the host.
+  # Returns true if it does, otherwise it adds a validation error and returns false.
+  def validate_association_taxonomy(association_name)
+    association = self.class.reflect_on_association(association_name)
+    raise ArgumentError, "Association #{association_name} not found" unless association
+    associated_object_id = public_send(association.foreign_key)
+    if Taxonomy.enabled_taxonomies.present? && associated_object_id.present? &&
+      Environment.with_taxonomy_scope(organization, location).find_by(id: associated_object_id).blank?
```

Comment:
This method name is generic but the implementation is tied to Environment.  Could this be refactored so it's usable for other associations, or otherwise change method name?

Reply:
Yes, the `Environment` was just a leftover from implementation, after trying to make it more generic. Fixed.